tests/rkt_exit_test.go

```diff
@@ -30,28 +71,82 @@ func TestExitCode(t *testing.T) {
 		ctx := newRktRunCtx()
 		defer ctx.cleanup()
 
-		cmd := fmt.Sprintf(`/bin/sh -c "`+
-			`%s --debug --insecure-skip-verify run --mds-register=false %s ;`+
-			`UUID=$(%s list --full|grep exited|awk '{print $1}') ;`+
-			`echo -n 'status=' ;`+
-			`%s status $UUID|grep '^app-rkt-inspect.*=[0-9]*$'|cut -d= -f2"`,
-			ctx.cmd(), imageFile,
-			ctx.cmd(),
-			ctx.cmd())
+		cmd := fmt.Sprintf(`%s --debug --insecure-skip-verify run --mds-register=false %s`,
+			ctx.cmd(), imageFile)
 		t.Logf("%s\n", cmd)
 		child, err := gexpect.Spawn(cmd)
 		if err != nil {
 			t.Fatalf("Cannot exec rkt")
 		}
-
-		err = expectWithOutput(child, fmt.Sprintf("status=%d", i))
+		err = child.Wait()
 		if err != nil {
-			t.Fatalf("Failed to get the status: %v", err)
+			t.Fatalf("rkt didn't terminate correctly: %v", err)
 		}
 
-		err = child.Wait()
+		checkStatus(t, ctx, "rkt-inspect", fmt.Sprintf("status=%d", i))
+	}
+}
+
+// TestExitCodeWithSeveralApps is testing a pod with three apps returning different
+// exit codes.
+func TestExitCodeWithSeveralApps(t *testing.T) {
+	image0File := patchTestACI("rkt-inspect-exit-0.aci", "--name=hello0",
+		"--exec=/inspect --print-msg=HelloWorld --exit-code=0")
+	defer os.Remove(image0File)
+
+	image1File := patchTestACI("rkt-inspect-exit-1.aci", "--name=hello1",
+		"--exec=/inspect --print-msg=HelloWorld --exit-code=1")
+	defer os.Remove(image1File)
+
+	image2File := patchTestACI("rkt-inspect-exit-2.aci", "--name=hello2",
+		"--exec=/inspect --print-msg=HelloWorld --exit-code=2 --sleep=1")
+	defer os.Remove(image2File)
+
+	ctx := newRktRunCtx()
+	defer ctx.cleanup()
+
+	cmd := fmt.Sprintf(`%s --debug --insecure-skip-verify run --mds-register=false %s %s %s`,
+		ctx.cmd(), image0File, image1File, image2File)
+	t.Logf("%s\n", cmd)
+	child, err := gexpect.Spawn(cmd)
+	if err != nil {
+		t.Fatalf("Cannot exec rkt")
+	}
+
+	for i := 0; i < 3; i++ {
+		// The 3 apps print the same message. We don't have any ordering
+		// guarantee but we don't need it.
+		err = expectTimeoutWithOutput(child, "HelloWorld", time.Minute)
 		if err != nil {
-			t.Fatalf("rkt didn't terminate correctly: %v", err)
+			t.Fatalf("Could not start the app (#%d): %v", i, err)
 		}
 	}
+
+	t.Logf("Check intermediary status\n")
+
+	// TODO: how to make sure hello0 and hello1 terminated? They should
+	// terminate soon because they already printed their HelloWorld message.
+	time.Sleep(100 * time.Millisecond)
+	checkStatus(t, ctx, "hello0", "status=0")
+	checkStatus(t, ctx, "hello1", "status=1")
+
+	// TODO: hello2 didn't terminate yet because of the sleep
+	// FIXME: running apps should not give any exit status, but they give 0
```

Comment:
It is actually fixed: apps don't give any exit status in `rkt status` when they are still running.

The comment was wrong: the app `hello2` should not be running anymore at this point. I updated the comment.